generators/getMetric.js

```diff
@@ -0,0 +1,88 @@
+const path = require('path')
+const fs = require('fs')
+const { metricsList } = require('./getMetricFunctions')
+const fsExtra = require('fs-extra')
+
+function generateGetMetric () {
+  let generatedDoc = ''
+  for (const metric of metricsList) {
+    generatedDoc += allTimeBoundTemplates(metric)
+  }
+  let file = path.join(`${__dirname}`, `../src/getMetric.js`)
+  if (fs.existsSync(file)) {
+    fsExtra.removeSync(file)
+  }
+  return fs.writeFileSync(file, generatedDoc)
+}
+
+function allTimeBoundTemplates (metric) {
+  if (!('currency' in metric)) {
+    metric.currency = ''
+  }
+  if (!('sheetMetricName' in metric)) {
+    metric.sheetMetricName = metric.metric
+  }
+  if (!('timeBound' in metric)) {
+    metric.timeBound = ''
+  }
+  let generatedString =
+    fillTemplate(metric.metric, metric.sheetMetricName, metric.description, metric.currency, metric.timeBound)
+
+  return generatedString
+}
+
+function fillTemplate (metric, sheetMetricName, description, currency, timeBound) {
+  let [options, currencyDescription, currencyArgument, timeBoundDescription, timeBoundArgument] =
+    optionsGenerator(currency, timeBound)
+  return `/**
+* Gets ${description}
+* @param {string} projectSlug Name of the asset at sanbase,
+* which can be found at the end of the URL (eg. the Santiment URL is
+* https://app.santiment.net/projects/santiment, so the projectSlug would be santiment).
+* @param {date} from The starting date to fetch the data. Example: DATE(2018, 9, 20)
+* @param {date} to The ending date to fetch the data. Example: DATE(2018, 9, 21)${currencyDescription}${timeBoundDescription}
+* @returns {number} returns ${description}.
+* @customfunction
+*/
+function SAN_${sheetMetricName.toUpperCase()} (projectSlug, from, to${currencyArgument}${timeBoundArgument}) {
+  return handleErrors_(getMetric_)('${metric}', projectSlug, from, to${options})
+}
+
+`
+}
+
+function optionsGenerator (currency, timeBound) {
```

Comment:
This function is not readable - I cannot really understand what is going on in there

Reply:
As I mentioned below:
- extract const to global consts
- reduce nesting of if/else clauses
- instead of appending to a string and doing extra work to know if you need a comma - add to list and join it
- etc.

Reply:
I think it is a lot readable now, I've reworked it as you asked.